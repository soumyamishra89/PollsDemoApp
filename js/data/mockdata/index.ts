export const mockQuestions = [
    {
        "question": "Favorite programming language?",
        "published_at": "2020-04-15T14:49:59.810110+00:00",
        "url": "/questions/5",
        "choices": [
            {
                "choice": "Objective-C",
                "votes": 0,
                "url": "/questions/5/choices/19"
            },
            {
                "choice": "Python",
                "votes": 0,
                "url": "/questions/5/choices/18"
            },
            {
                "choice": "Ruby",
                "votes": 0,
                "url": "/questions/5/choices/20"
            },
            {
                "choice": "Swift",
                "votes": 0,
                "url": "/questions/5/choices/17"
            }
        ],
        "totalVotes": 0
    },
    {
        "question": "Capital of Germany?",
        "published_at": "2020-04-15T11:00:59.542382+00:00",
        "url": "/questions/4",
        "choices": [
            {
                "choice": "Berlin",
                "votes": 0,
                "url": "/questions/4/choices/13"
            },
            {
                "choice": "Frankfurt",
                "votes": 0,
                "url": "/questions/4/choices/16"
            },
            {
                "choice": "Hamburg",
                "votes": 0,
                "url": "/questions/4/choices/14"
            },
            {
                "choice": "Munich",
                "votes": 0,
                "url": "/questions/4/choices/15"
            }
        ],
        "totalVotes": 0
    },
    {
        "question": "test q?",
        "published_at": "2020-04-15T09:59:32.552020+00:00",
        "url": "/questions/3",
        "choices": [
            {
                "choice": "Objective-C",
                "votes": 0,
                "url": "/questions/3/choices/11"
            },
            {
                "choice": "Python",
                "votes": 0,
                "url": "/questions/3/choices/10"
            },
            {
                "choice": "Ruby",
                "votes": 0,
                "url": "/questions/3/choices/12"
            },
            {
                "choice": "Swift",
                "votes": 0,
                "url": "/questions/3/choices/9"
            }
        ],
        "totalVotes": 0
    },
    {
        "question": "",
        "published_at": "2020-04-10T16:05:25.127319+00:00",
        "url": "/questions/2",
        "choices": [
            {
                "choice": "Python",
                "votes": 14,
                "url": "/questions/2/choices/6"
            },
            {
                "choice": "Ruby",
                "votes": 6,
                "url": "/questions/2/choices/8"
            },
            {
                "choice": "Objective-C",
                "votes": 2,
                "url": "/questions/2/choices/7"
            },
            {
                "choice": "Swift",
                "votes": 1,
                "url": "/questions/2/choices/5"
            }
        ],
        "totalVotes": 23
    },
    {
        "question": "Favourite programming language?",
        "published_at": "2020-04-09T03:28:20.545996+00:00",
        "url": "/questions/1",
        "choices": [
            {
                "choice": "Python",
                "votes": 34,
                "url": "/questions/1/choices/2"
            },
            {
                "choice": "Objective-C",
                "votes": 8,
                "url": "/questions/1/choices/3"
            },
            {
                "choice": "Ruby",
                "votes": 4,
                "url": "/questions/1/choices/4"
            },
            {
                "choice": "Swift",
                "votes": 3,
                "url": "/questions/1/choices/1"
            }
        ],
        "totalVotes": 49
    }
]
