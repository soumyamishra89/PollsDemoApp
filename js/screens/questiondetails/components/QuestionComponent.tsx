import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Paragraph, Title } from 'react-native-paper';

interface Props {
    question: string,
    publishedDate: string
}
export default function QuestionComponent(props: Props) {
    return (
        <View style={styles.questionContainer}>
            <Title>{props.question}</Title>
            <Paragraph>{props.publishedDate}</Paragraph>
        </View>
    )
}

const styles = StyleSheet.create({
    questionContainer: {
        width: '100%',
        marginVertical: 16
    }
})