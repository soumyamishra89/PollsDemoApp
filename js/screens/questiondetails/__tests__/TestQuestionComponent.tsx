import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import QuestionComponent from '../components/QuestionComponent';
import { mockQuestions } from '../../../data/mockdata';

describe("QuestionComponent", () => {
    describe("Snapshot test", () => {
        it("With question data", () => {
            const pollQuestion = mockQuestions[0];
            const questionTree = renderer.create(<QuestionComponent question={pollQuestion.question} publishedDate={pollQuestion.published_at} />).toJSON();
            expect(questionTree).toMatchSnapshot();

        })
    })
})