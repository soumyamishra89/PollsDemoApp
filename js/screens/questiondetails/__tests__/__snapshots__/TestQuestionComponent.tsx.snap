// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionComponent Snapshot test With question data 1`] = `
<View
  style={
    Object {
      "marginVertical": 16,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            undefined,
          ],
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Favorite programming language?
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 14,
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            undefined,
          ],
          Array [
            Object {
              "fontSize": 14,
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            undefined,
          ],
        ],
      ]
    }
  >
    2020-04-15T14:49:59.810110+00:00
  </Text>
</View>
`;
